cards/tests/dominion/test_moneylender.py

```diff
@@ -0,0 +1,79 @@
+from cards.kingdom_cards.dominion import Moneylender
+from games.factories import GameFactory
+from players.factories import PlayerFactory
+from testing import BaseTestCase
+from turns.factories import AdHocTurnFactory, TurnFactory
+
+
+class MoneylenderCardTestCase(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.player = PlayerFactory()
+        self.game = GameFactory(players=[self.player] + PlayerFactory.create_batch(11))
+        self.turn = TurnFactory(player=self.player, game=self.game)
+        self.deck = self.game.decks.get(player=self.player)
+        self.card = Moneylender()
+
+    def test_perform_specific_action(self):
+        with self.assert_adhoc_turn_created():
+            adhoc_turn = self.card.perform_specific_action(
+                deck=self.deck,
+                turn=self.turn,
+            )
+        self.assert_adhoc_turn(
+            adhoc_turn=adhoc_turn,
+            turn=self.turn,
+            player=self.player,
+            game=self.game,
+            card=self.card,
+        )
+
+    def test_perform_specific_action_no_copper(self):
+        self.deck.hand = ['Estate'] * 4
+        with self.assert_adhoc_turn_not_created():
+            self.card.perform_specific_action(
+                deck=self.deck,
+                turn=self.turn,
+            )
+
+
+class MoneylenderFormTestCase(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.player = PlayerFactory()
+        self.game = GameFactory(players=[self.player] + PlayerFactory.create_batch(11))
+        self.turn = TurnFactory(player=self.player, game=self.game)
+        self.deck = self.game.decks.get(player=self.player)
+        self.card = Moneylender()
+        self.adhoc_turn = AdHocTurnFactory(
+            player=self.player,
+            game=self.game,
+            turn=self.turn,
+            card=self.card,
+        )
+
+    def test_decline_trashing(self):
+        trash = self.game.trash_pile[:]
+        hand = self.deck.hand[:]
+        money = self.turn.available_money
+        form = self.build_card_form(adhoc_turn=self.adhoc_turn, selection='1')
```

Comment:
We should pull `'1'` and `'0'` into variables and dump them on the base test case. Something like `selection_yes = '0'` and `selection_no = '1'`. I'm mostly pitching this because it looks like 0 is yes and 1 is no. Which by the way is the opposite of what it likely ought to be. Lol. By pulling all these out into variables, we can switch them to whatever we want and only need to update in once place.

Reply:
Agreed. Not only are 'Yes' and 'No' incorrect in terms of boolean, they also change depending on which form we're in. I will leave this PR unmerged to ensure I remember to do that.